src/MongoDB.Driver.Core/Core/NativeLibraryLoader/RelativeLibraryLocatorBase.cs

```diff
@@ -35,31 +36,39 @@ public virtual Assembly GetLibraryBaseAssembly()
 
         public virtual string GetLibraryBasePath()
         {
-            var assembly = GetLibraryBaseAssembly();
-            var codeBase = assembly.CodeBase;
-            var uri = new Uri(codeBase);
-            var absolutePath = uri.AbsolutePath;
-            return Path.GetDirectoryName(absolutePath);
+            // In the nuget package, get the shared library from a relative path of this assembly
+            // Also, when running locally, get the shared library from a relative path of this assembly
+            var assembly = typeof(LibraryLoader).GetTypeInfo().Assembly;
+            var location = assembly.Location;
+            string basePath = Path.GetDirectoryName(location);
+            return Path.GetDirectoryName(basePath);
         }
 
         public abstract string GetLibraryRelativePath(SupportedPlatform currentPlatform);
 
         // private methods
-        private string FindLibraryOrThrow(string basePath, string relativePath)
+        private string FindLibraryOrThrow(string[] basePaths, string[] suffixPaths, string library)
```

Comment:
Searching multiple locations so it'll work inside unit tests as well as when being consumed as a NuGet package

Reply:
sounds like the main change here is adding an empty string as path where we're trying to find the library. Still checking other changes for now.

Reply:
This feels like a bandaid. If the snappy64.dll were in the right place to begin with we wouldn't have to be searching all over the place for it.